plugin/src/main/java/net/codingarea/challenges/plugin/challenges/implementation/goal/forcebattle/ExtremeForceBattleGoal.java

```diff
@@ -0,0 +1,291 @@
+package net.codingarea.challenges.plugin.challenges.implementation.goal.forcebattle;
+
+import net.anweisen.utilities.bukkit.utils.animation.SoundSample;
+import net.anweisen.utilities.bukkit.utils.misc.BukkitReflectionUtils;
+import net.anweisen.utilities.common.config.Document;
+import net.codingarea.challenges.plugin.ChallengeAPI;
+import net.codingarea.challenges.plugin.challenges.implementation.goal.forcebattle.targets.*;
+import net.codingarea.challenges.plugin.challenges.type.abstraction.ForceBattleDisplayGoal;
+import net.codingarea.challenges.plugin.challenges.type.helper.ChallengeHelper;
+import net.codingarea.challenges.plugin.content.Message;
+import net.codingarea.challenges.plugin.content.Prefix;
+import net.codingarea.challenges.plugin.management.scheduler.policy.TimerPolicy;
+import net.codingarea.challenges.plugin.management.scheduler.task.ScheduledTask;
+import net.codingarea.challenges.plugin.utils.item.ItemBuilder;
+import net.codingarea.challenges.plugin.utils.misc.InventoryUtils;
+import net.codingarea.challenges.plugin.utils.misc.NameHelper;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.World;
+import org.bukkit.advancement.AdvancementProgress;
+import org.bukkit.block.Biome;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.entity.EntityDamageEvent;
+import org.bukkit.event.entity.EntityDeathEvent;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+
+/**
+ * @author sehrschlechtYT | https://github.com/sehrschlechtYT
+ * @since 2.2.3
+ */
+public class ExtremeForceBattleGoal extends ForceBattleDisplayGoal<ForceTarget<?>> {
+
+
+    public ExtremeForceBattleGoal() {
+        super(Message.forName("menu-extreme-force-battle-goal-settings"));
+
+        registerSetting("give-item", new BooleanSubSetting(
+                () -> new ItemBuilder(Material.CHEST, Message.forName("item-force-item-battle-goal-give-item")),
+                false
+        ));
+        registerSetting("give-block", new BooleanSubSetting(
+                () -> new ItemBuilder(Material.CHEST, Message.forName("item-force-block-battle-goal-give-block")),
+                false
+        ));
+    }
+
+    @NotNull
+    @Override
+    public ItemBuilder createDisplayItem() {
+        return new ItemBuilder(Material.BOOK, Message.forName("item-extreme-force-battle-goal"));
+    }
+
+    @Override
+    protected ForceTarget<?>[] getTargetsPossibleToFind() {
+        //Currently not needed as the setRandomTarget logic of the ForceBattleGoal class is not used
```

Comment:
Then what is it useful for?

Reply:
The method is overriden because it is required to override it as it is an abstract method in the parent (`ForceBattleGoal`) class.
The Extreme Force Battle Goal has its own `setRandomTarget` logic, and therefore the `getTargetsPossibleToFind` method is not used.